### PR TITLE
Decrease the block time threshold (3x)

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo/issues/1232

### CODE REVIEW

**Comment by shargon** [comment]
What do you propose?

File: neo/Consensus/ConsensusService.cs [hunk 1]
```diff
@@ -405,7 +405,7 @@ private void OnPrepareRequestReceived(ConsensusPayload payload, PrepareRequest m
             if (context.RequestSentOrReceived || context.NotAcceptingPayloadsDueToViewChanging) return;
             if (payload.ValidatorIndex != context.Block.ConsensusData.PrimaryIndex || message.ViewNumber != context.ViewNumber) return;
             Log($"{nameof(OnPrepareRequestReceived)}: height={payload.BlockIndex} view={message.ViewNumber} index={payload.ValidatorIndex} tx={message.TransactionHashes.Length}");
-            if (message.Timestamp <= context.PrevHeader.Timestamp || message.Timestamp > TimeProvider.Current.UtcNow.AddMinutes(10).ToTimestampMS())
+            if (message.Timestamp <= context.PrevHeader.Timestamp || message.Timestamp > TimeProvider.Current.UtcNow.AddMinutes(2).ToTimestampMS())
```
Comment (shargon) [inline neo/Consensus/ConsensusService.cs]: What do you think about `(Validators.Length + 1) * TimePerBlock` ?

Comment (lock9) [reply neo/Consensus/ConsensusService.cs]: I prefer dynamic value than a hardcoded one.

**Comment by shargon** [comment]
@lock9 The 2 minutes it's acording to the flow that i showed in the issue #1232, but I prefer to adjust it according to the time per block and the validators

**Comment by lock9** [comment]
@shargon  what would be the value for current network conditions?

**Comment by shargon** [comment]
with my formula, 2 minutes. (8*15)

**Comment by erikzhang** [comment]
Why is it related to the number of validators?

**Comment by shargon** [comment]
I thought that is better to have a number related because if there are 4 validators, and one evil, with 2 minutes he can do the attack.

File: neo/Consensus/ConsensusContext.cs [hunk 2]
```diff
@@ -349,6 +350,7 @@ public void Reset(byte viewNumber)
                 };
                 var pv = Validators;
                 Validators = NativeContract.NEO.GetNextBlockValidators(Snapshot);
+                PrepareRequestThreshold = TimeSpan.FromMilliseconds((Validators.Length + 1) * Blockchain.TimePerBlock.TotalMilliseconds);
```
Comment (erikzhang) [inline neo/Consensus/ConsensusContext.cs]: What about `8 * Blockchain.TimePerBlock.TotalMilliseconds`? I can't see any relationship between the number and the threshold.

Comment (shargon) [reply neo/Consensus/ConsensusContext.cs]: My rationale is about how much time a malicious node can earn per round, so it depends the number of the CN.

Comment (erikzhang) [reply neo/Consensus/ConsensusContext.cs]: https://github.com/neo-project/neo/blob/72be7f6a26f3ed222f1ad5dc1b4391c62fa4f67d/neo/Consensus/ConsensusService.cs#L408

But it is compared to the current time, not the current block time. So you can't earn time from every round.

Comment (shargon) [reply neo/Consensus/ConsensusContext.cs]: Agree, so.. What do you prefer: fix it to 2 minutes, or 8 * blockTime?

Comment (erikzhang) [reply neo/Consensus/ConsensusContext.cs]: Both are good to me.

**Comment by vncoelho** [comment]
It looks good like this.

File: neo/Consensus/ConsensusService.cs [hunk 3]
```diff
@@ -29,6 +29,7 @@ internal class Timer { public uint Height; public byte ViewNumber; }
         private ICancelable timer_token;
         private DateTime block_received_time;
         private bool started = false;
+        public readonly TimeSpan PrepareRequestThreshold = TimeSpan.FromMilliseconds(8 * Blockchain.TimePerBlock.TotalMilliseconds);
```
Comment (erikzhang) [inline neo/Consensus/ConsensusService.cs]:
```suggestion
        public static readonly TimeSpan PrepareRequestThreshold = 8 * Blockchain.TimePerBlock;
```

**Pull request merged**